lib/libc/stdlib/malloc/malloc_revoke.3

```diff
@@ -29,38 +34,96 @@
 .\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 .\" SUCH DAMAGE.
 .\"
-.Dd May 14, 2021
-.Dt MALLOC_REVOKE 3
+.Dd January 31, 2025
+.Dt MALLOC_QUARANTINE 3
 .Os
 .Sh NAME
-.Nm malloc_revoke
-.Nd trigger a revocation pass
+.Nm malloc_revoke_enabled ,
+.Nm malloc_revoke_quarantine_force_flush ,
+.Nm malloc_revoke ,
+.Nm malloc_is_revoking
+.Nd malloc revocation status and control
 .Sh LIBRARY
 .Lb libc
 .Sh SYNOPSIS
+.In sys/errno.h
 .In stdlib.h
+.Ft int
+.Fn malloc_revoke_quarantine_force_flush "void"
 .Ft void
 .Fn malloc_revoke "void"
+.In malloc_np.h
+.Ft bool
+.Fn malloc_revoke_enabled "void"
+.Ft bool
+.Fn malloc_is_revoking "void"
 .Sh DESCRIPTION
-On systems supporting revocation,
-.Fn
-triggers a full revocation pass, returning when the pass is complete.
-This results in the revocation of all pointers staged for quarantine.
-Depending on their implementation, this may be a subset of pointers
-previously passed to
+On systems which support help temporal safety though a quarantine and
+sweep mechanism (CHERI systems) these interfacees allow status to be
+obtained and quarantine controlled.
```

Comment:
This reads a bit odd, maybe "allow status to be obtained and quarantine to be controlled" to be consistent, but that's a lot of passive voice.  Maybe:

```
On systems which support heap temporal safety through a quarantine and
sweep mechanism (such as CHERI systems), these interfaces permit
control and monitoring of quarantined heap memory.
On other systems, these interfaces are stubs.
```